packages/verified-fetch/src/plugins/plugin-handle-dag-pb.ts

```diff
@@ -94,9 +95,22 @@ export class DagPbPlugin extends BasePlugin {
         path = rootFilePath
         resolvedCID = entry.cid
       } catch (err: any) {
+        this.log.error('error loading path %c/%s', dirCid, rootFilePath, err)
         options?.signal?.throwIfAborted()
-        log('error loading path %c/%s', dirCid, rootFilePath, err)
-        return notSupportedResponse('Unable to find index.html for directory at given path. Support for directories with implicit root is not implemented')
+        context.isDirectory = true
+        context.directoryEntries = []
+        this.log.trace('attempting to get directory entries because index.html was not found')
+        const fs = unixfs({ ...helia, blockstore: getBlockstore(context.cid, context.resource, options?.session ?? true, options) })
+        try {
+          for await (const dirItem of fs.ls(dirCid, { signal: options?.signal, onProgress: options?.onProgress })) {
+            context.directoryEntries.push(dirItem)
+          }
+          // dir-index-html plugin or dir-index-json (future idea?) plugin should handle this
+          return null
+        } catch (e) {
+          log.error('error listing directory %c', dirCid, e)
+          return notSupportedResponse('Unable to get directory contents')
+        }
```

Comment:
this could be moved to dir-index-html plugin but I thought it made the most sense here.